geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSet.java

```diff
@@ -69,6 +69,30 @@ public RedisSet(int expectedSize) {
    */
   public RedisSet() {}
 
+  public static int smove(RegionProvider regionProvider, List<RedisKey> keys, byte[] member) {
+    Region<RedisKey, RedisData> region = regionProvider.getDataRegion();
+    RedisKey sourceKey = keys.get(0);
+    RedisKey destKey = keys.get(1);
+
+    RedisSet source = regionProvider.getTypedRedisData(REDIS_SET, sourceKey, false);
+    if (source.scard() == 0 || !source.members.contains(member)) {
+      return 0;
+    }
+
+    RedisSet destination = regionProvider.getTypedRedisData(REDIS_SET, keys.get(1), false);
+
+    List<byte[]> movedMember = new ArrayList<>();
+    movedMember.add(member);
+    source.srem(movedMember, region, sourceKey);
+    if (destination.scard() == 0) {
+      region.put(destKey, new RedisSet(movedMember));
+    } else {
+      destination.sadd(movedMember, region, destKey);
+    }
```

Comment:
Because of the implementation of `sadd()` in `NullRedisSet`, this if/else is not necessary. Just calling `destination.sadd(movedMember, region, destKey);` will result in the correct behaviour in both cases.